util/zenfs.cc

```diff
@@ -69,21 +70,56 @@ Status zenfs_mount(std::unique_ptr<ZonedBlockDevice> &zbd,
   return s;
 }
 
+// Create or check pre-existing aux directory and fail if it is
+// inaccessible by current user and if it has previous data
+int create_aux_path() {
+  struct dirent *dent;
+  DIR *aux_dir;
+  unsigned int nfiles = 0;
+  int ret = 0;
+
+  errno = 0;
+  ret = mkdir(FLAGS_aux_path.c_str(), S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH);
+  if (ret < 0 && EEXIST != errno) {
+    fprintf(stderr, "ZenFS: Error creating directory %s: %s\n",
+            FLAGS_aux_path.c_str(), strerror(errno));
+    return 1;
+  }
+  // The aux_path is now available, check if it is not empty and
```

Comment:
One more corner case here.
What happens if `aux_path` exists but it is a file rather than a directory.
Another question is a `symlink` pointing to either directory (`OK`) or file (`not OK`).

Reply:
Done.